core/src/main/scala/org/apache/spark/resource/ResourceProfileManager.scala

```diff
@@ -34,7 +34,7 @@ import org.apache.spark.util.Utils.isTesting
  */
 @Evolving
 private[spark] class ResourceProfileManager(sparkConf: SparkConf) extends Logging {
-  private val resourceProfileIdToResourceProfile = new ConcurrentHashMap[Int, ResourceProfile]()
+  private val resourceProfileIdToResourceProfile = new HashMap[Int, ResourceProfile]()
```

Comment:
Is there a reason to change like this?

Reply:
yep changed the synchronization due to concurrentHashMap not being friendly for the new function to find existing profiles.